src/app/core/market/api/bid/bid.service.ts

```diff
@@ -20,12 +20,12 @@ export class BidService {
   order(cart: Cart, profile: any): Observable<boolean> {
     let nBidsPlaced = 0;
     const addressIdOfProfile: number = profile.ShippingAddresses[0].id;
-
+    const profileId: string = profile.ShippingAddresses[0].profileId;
     return new Observable((observer) => {
       cart.listings.forEach((listing: Listing) => {
         if (listing.hash) {
           // bid for item
-          this.market.call('bid', ['send', listing.hash, addressIdOfProfile])
+          this.market.call('bid', ['send', listing.hash, addressIdOfProfile, profileId])
```

Comment:
the order is wrong..
It should be
`[send, listing_hash, profile_id, address_id]`

Also please just pass through `profile.ShippingAddresses[0].id` and `profile.Id` so that we're not duplicating variables unnecessarily at this point in time